src/hotspot/share/gc/shared/memAllocator.cpp

```diff
@@ -406,13 +406,28 @@ oop ObjArrayAllocator::initialize(HeapWord* mem) const {
   assert(_length >= 0, "length should be non-negative");
   if (_do_zero) {
     mem_clear(mem);
+    mem_zap_alignment_padding(mem);
```

Comment:
I feel the ordering here is a bit odd; these methods are for spaces corresponding to before-body, body and after-body.

Reply:
Yeah issue is that `mem_clear(mem);` will clear the padding. Maybe I should do the same for `ZObjArrayAllocator` as well. To keep them consistent. Even though the `ZObjArrayAllocator` does not clear that memory. 

Also this is debug code, maybe just doing it after the normal clearing is the better choice. So regardless how much memory the normal clearing does we always end up with the padding zapped.